pymeasure/instruments/validators.py

```diff
@@ -72,6 +72,31 @@ def truncated_range(value, values):
         return min(values)
 
 
+def modular_range(value, values):
```

Comment:
I think `modular_range` and `modular_range_bidirectional` are great additions! I think they deserve tests to ensure that they continue to work as planned. These should happen for both classes in https://github.com/ralph-group/pymeasure/blob/master/tests/instruments/test_validators.py, following the other tests as examples.

Reply:
Done. I included `isclose` in the validator testing file, which should be in the standard math library in Python version 3.5+ (https://docs.python.org/3/whatsnew/3.5.html#pep-485-a-function-for-testing-approximate-equality). This seemed necesary since I was getting errors due to floating point imprecision using Python's built-in modulo operator.

Reply:
Note that pytest nowadays has an assert_almost_equal statement (or is it also called is close? I'm on the phone so can't easily check)

Reply:
@bilderbuchi raised a great suggestion to use pytest directly. @jmittelstaedt, check out the `approx` function from pytest (https://docs.pytest.org/en/latest/reference.html#pytest-approx). I would suggest using that instead, because that should satisfy tests on Python 3.4 as well.

Reply:
Thanks for pointing that out! I've changed the tests to use approx instead.

Reply:
Apparently approx is also only available in a more recent version of python (or pytest at least). I'm now just checking what should be the simplest way by using `abs(validator(x) - x) < tol`. One of the Travis tests still failed, but I couldn't understand what the error is trying to tell me (or, I couldn't understand how I could have caused an error like that with what I've done)

Reply:
approx is available from pytest 3.0 (so you have to edit the entry in setup.py to reflect that). 3.0 is available for py3.4+, so that won't be a problem. (Note the API [changed a little in pytest 3.2](https://docs.pytest.org/en/latest/reference.html#pytest-approx) so you can't use  >, >=, < and <= anymore)

Reply:
I don't know for sure, but I don't think the CI error is related to this PR, but @cjermain will be better able to judge that.

Reply:
It looks like a Qt error, which is not related to these changes. I will merge and we can revisit.